### ISSUE TITLE
Game crash after press 'U' or /cam start - 1.20.4 with CMDCam 2.1.1

### ISSUE THREAD

**Issue by TrangggP** [CreativeMD/CMDCam#104]
[crash-2024-09-18_17.48.09-client.txt](https://github.com/user-attachments/files/17042752/crash-2024-09-18_17.48.09-client.txt)

```
---- Minecraft Crash Report ----
// Don't do that.

Time: 2024-09-18 17:48:09
Description: Unexpected error

java.lang.IllegalStateException: Trying to return to in-game GUI during disconnection
	at net.minecraft.client.Minecraft.setScreen(Minecraft.java:1092) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.MouseHandler.grabMouse(MouseHandler.java:342) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:creativecore.mixins.json:MouseHandlerAccessor from mod creativecore,pl:mixin:A,pl:runtimedistcleaner:A}
	at team.creative.cmdcam.common.scene.mode.DefaultMode.process(DefaultMode.java:27) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:classloading}
	at team.creative.cmdcam.common.scene.run.CamRun.renderTick(CamRun.java:134) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:classloading,pl:runtimedistcleaner:A}
	at team.creative.cmdcam.common.scene.CamScene.renderTick(CamScene.java:199) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:classloading,re:mixin}
	at team.creative.cmdcam.client.CMDCamClient.renderTickPath(CMDCamClient.java:225) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:mixin,re:classloading}
	at team.creative.cmdcam.client.CamEventHandlerClient.onRenderTick(CamEventHandlerClient.java:153) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:runtimedistcleaner:A}
	at net.neoforged.bus.EventBus.post(EventBus.java:386) ~[bus-7.2.0.jar%23174!/:?] {}
	at net.neoforged.bus.EventBus.post(EventBus.java:351) ~[bus-7.2.0.jar%23174!/:?] {}
	at net.neoforged.neoforge.event.EventHooks.onRenderTickStart(EventHooks.java:814) ~[neoforge-20.4.237-universal.jar%23182!/:?] {re:classloading}
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1261) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.updateScreenAndTick(Minecraft.java:2298) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.disconnect(Minecraft.java:2232) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.multiplayer.ClientCommonPacketListenerImpl.onDisconnect(ClientCommonPacketListenerImpl.java:202) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.network.Connection.handleDisconnection(Connection.java:571) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading}
	at net.minecraft.client.multiplayer.MultiPlayerGameMode.tick(MultiPlayerGameMode.java:277) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.tick(Minecraft.java:1887) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1229) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.run(Minecraft.java:818) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:244) ~[1.20.4-20.4.237.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.client.Minecraft.setScreen(Minecraft.java:1092) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.MouseHandler.grabMouse(MouseHandler.java:342) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:creativecore.mixins.json:MouseHandlerAccessor from mod creativecore,pl:mixin:A,pl:runtimedistcleaner:A}
	at team.creative.cmdcam.common.scene.mode.DefaultMode.process(DefaultMode.java:27) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:classloading}
	at team.creative.cmdcam.common.scene.run.CamRun.renderTick(CamRun.java:134) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:classloading,pl:runtimedistcleaner:A}
	at team.creative.cmdcam.common.scene.CamScene.renderTick(CamScene.java:199) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:classloading,re:mixin}
	at team.creative.cmdcam.client.CMDCamClient.renderTickPath(CMDCamClient.java:225) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:mixin,re:classloading}
	at team.creative.cmdcam.client.CamEventHandlerClient.onRenderTick(CamEventHandlerClient.java:153) ~[CMDCam_FORGE_v2.1.1_mc1.20.4.jar%23179!/:2.1.1] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:runtimedistcleaner:A}
	at net.neoforged.bus.EventBus.post(EventBus.java:386) ~[bus-7.2.0.jar%23174!/:?] {}
	at net.neoforged.bus.EventBus.post(EventBus.java:351) ~[bus-7.2.0.jar%23174!/:?] {}
	at net.neoforged.neoforge.event.EventHooks.onRenderTickStart(EventHooks.java:814) ~[neoforge-20.4.237-universal.jar%23182!/:?] {re:classloading}
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1261) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.updateScreenAndTick(Minecraft.java:2298) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.disconnect(Minecraft.java:2232) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.multiplayer.ClientCommonPacketListenerImpl.onDisconnect(ClientCommonPacketListenerImpl.java:202) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.network.Connection.handleDisconnection(Connection.java:571) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading}
	at net.minecraft.client.multiplayer.MultiPlayerGameMode.tick(MultiPlayerGameMode.java:277) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:runtimedistcleaner:A}
-- Uptime --
Details:
	JVM uptime: 74.355s
	Wall uptime: 43.328s
	High-res time: 69.048s
	Client ticks: 670 ticks / 33.500s
Stacktrace:
	at net.minecraft.client.Minecraft.fillReport(Minecraft.java:2482) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.emergencySaveAndCrash(Minecraft.java:941) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.run(Minecraft.java:839) ~[client-1.20.4-20231207.154220-srg.jar%23181!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:244) ~[1.20.4-20.4.237.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}


-- Affected level --
Details:
	All players: 3 total; [LocalPlayer['{MINECRAFT_USERNAME}'/384, l='ClientLevel', x=-1.90, y=64.00, z=-4.53], RemotePlayer['CIT-09f1055d9fff'/28, l='ClientLevel', x=0.54, y=65.00, z=-14.50], RemotePlayer['§a§lSurvival'/20, l='ClientLevel', x=3.52, y=65.00, z=-14.62]]
	Chunk stats: 225, 77
	Level dimension: minecraft:overworld
	Level spawn location: World: (0,65,0), Section: (at 0,1,0 in 0,4,0; chunk contains blocks 0,-64,0 to 15,319,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 156291 game time, 108252291 day time
	Server brand: langbangvn.net
	Server type: Non-integrated multiplayer server
	Tracked entity count: 11

```

**Comment by TrangggP** [comment]
Fix it: I just downrage to minecraft 1.20.1 and it works 🤔

**Comment by CreativeMD** [comment]
Yeah, sorry the version for 1.20.4 is not supported anymore. Please use 1.20.1 or 1.21.
